Libraries/LibC/netdb.cpp

```diff
@@ -84,7 +84,9 @@ static int connect_to_lookup_server()
 
     sockaddr_un address;
     address.sun_family = AF_LOCAL;
-    strlcpy(address.sun_path, "/tmp/portal/lookup", sizeof(address.sun_path));
+    bool fits = String("/tmp/portal/lookup").copy_characters_to_buffer(address.sun_path, sizeof(address.sun_path));
+    // The (constant) address is short enough to always fit into the buffer.
+    ASSERT(fits);
```

Comment:
This is kinda weird, why not a compiletime assert?
Probably something like
```c++
constexpr static auto lookup_path = "/tmp/portal/lookup";
static_assert(__builtin_strlen(lookup_path) < sizeof(declval<sockaddr_un>().sun_path),
    "sockaddr_un::sun_path not large enough, or lookup_path invalid?");
__builtin_memcpy(address.sun_path, lookup_path, __builtin_strlen(lookup_path));
```
, even as a template inline function somewhere so it could be reused?

Reply:
:thinking: perhaps even

```cpp
address = {
    .sun_path = "/tmp/portal/lookup"
};
```

Lemme see if that works.

Reply:
I doubt it'd be happy with that...

Reply:
It works!

```cpp
../Libraries/LibC/netdb.cpp: In function 'int connect_to_lookup_server()':
../Libraries/LibC/netdb.cpp:87:9: error: initializer-string for 'char [108]' is too long [-fpermissive]
   87 |         "/tmp/portal/lookup/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```

Reply:
Noice!

Reply:
I don't seem to be able to convince the compiler about the ping buffer thing though...

`error: C99 designator 'msg' outside aggregate initializer`

Dang it, C++.

So I'll leave that one as an assertion.

Reply:
Perhaps the ping message class has a ctor?
Ah well. 1/2 is still good; though I'll push a little more for a compiletime array copy thingy